src/js/components/modals/ImageViewerModal.js

```diff
@@ -38,9 +38,24 @@ class ImageViewerModal extends React.Component {
     }
   }
 
+  componentWillUpdate(nextProps, {isLoadingImage}) {
+    let {images, selectedImage} = this.props;
+    let nextImages = nextProps.images;
+    let nextSelectedImage = nextProps.selectedImage;
+    // Two images in a row are the same, img.onLoad will never be called, so
+    // here we are updating manually
+    if (isLoadingImage &&
+        images[selectedImage] === nextImages[nextSelectedImage]) {
+      this.handleLoadingImageChange(false);
```

Comment:
I just thought of something. I'm pretty sure if you add a `key` property on the `img` tag and make it different for every single image (including the same image), the `onLoad` will trigger again and we can evade all of this